internal/postprocessor/main.go

```diff
@@ -268,3 +246,177 @@ func docURL(cloudDir, importPath string) (string, error) {
 	pkgPath := strings.TrimPrefix(strings.TrimPrefix(importPath, mod), "/")
 	return "https://cloud.google.com/go/docs/reference/" + mod + "/latest/" + pkgPath, nil
 }
+
+func (c *config) AmendPRDescription(ctx context.Context) error {
+	log.Println("Amending PR title and body")
+	pr, err := c.getPR(ctx)
+	if err != nil {
+		return err
+	}
+	prTitle := pr.Title
+	prBody := pr.Body
+	newPRTitle, newPRBody, err := c.processCommit(*prTitle, *prBody)
+	if err != nil {
+		return err
+	}
+	if err := c.writePRCommitToFile(newPRTitle, newPRBody); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *config) processCommit(title, body string) (string, string, error) {
+	var newPRTitle string
+	var commitTitle string
+	var commitTitleIndex int
+
+	bodySlice := strings.Split(body, "\n")
+	for index, line := range bodySlice {
+		if strings.Contains(line, "[REPLACEME]") {
+			commitTitle = line
+			commitTitleIndex = index
+			continue
+		}
+		if !strings.Contains(line, "googleapis/googleapis/") {
+			continue
+		}
+		commitPkg, err := c.getScope(line)
+		if err != nil {
+			return "", "", err
+		}
+
+		if newPRTitle == "" {
+			newPRTitle = updateCommitTitle(title, commitPkg)
+			continue
+		}
+		newCommitTitle := updateCommitTitle(commitTitle, commitPkg)
+		bodySlice[commitTitleIndex] = newCommitTitle
+	}
+	body = strings.Join(bodySlice, "\n")
+	return newPRTitle, body, nil
+}
+
+func (c *config) getPR(ctx context.Context) (*github.PullRequest, error) {
+	client := github.NewClient(nil)
+	prs, _, err := client.PullRequests.List(ctx, c.githubUsername, "google-cloud-go", nil)
+	if err != nil {
+		return nil, err
+	}
+	pr, err := c.findValidPR(ctx, prs)
+	if err != nil {
+		return nil, err
+	}
+	return pr, nil
+}
+
+func (c *config) findValidPR(ctx context.Context, prs []*github.PullRequest) (*github.PullRequest, error) {
```

Comment:
nit: I think I would update this function to not return an error. It could just return nil if it did not find one. Or maybe this could even be inlined into the calling function.

Reply:
Done, but not sure my change is in the spirit of your intent. See getPR func above where I have added a check: `if pr == nil {return nil, errors.New("no PR found")}

Reply:
Done.